greeter/README.md

```diff
@@ -1,21 +1,706 @@
+   const httpReq = await invocationCtx.get(RestBindings.Http.REQUEST, {
+     optional: true,
+   });
+   /* istanbul ignore if */
+   // A workaround to skip the explorer
+   if (EXCLUDED_PATHS.includes(httpReq.path)) {
+     return next();
+   }
+   const key = httpReq.path;
+   const lang = httpReq.acceptsLanguages(['en', 'zh']) || 'en';
+   const cachingKey = `${lang}:${key}`;
+   const cachedResult = await this.cachingService.get(cachingKey);
+   if (cachedResult) {
+     console.error('Cache found for %s %j', cachingKey, cachedResult);
+     return cachedResult;
+   }
+
+   const result = await next();
+
+   // Add post-invocation logic here
+   await this.cachingService.set(cachingKey, result);
+   return result;
+   ```
+
+4. Then bind Observer and Interceptor to you app
+   in `src/application.ts` file, add these lines:
+
+   ```ts
+   this.bind(RestExplorerBindings.CONFIG).to({
+     path: '/explorer',
+   });
+   // bind interceptor and service to app
+   this.add(createBindingFromClass(CachingService, {key: CACHING_SERVICE}));
+   this.add(createBindingFromClass(CachingInterceptor));
+
+   this.component(RestExplorerComponent);
+   this.component(GreetingComponent);
+   ```
+
+### Step 4: Try it out again!
+
+Let's try it out to see the caching server and the observer that we just bind. Restart your application.
+
+#### Test cache
+
+The expire time for each result is set to 10s (10000 milliseconds by default).
+To test the cached result and its expiration, you can
+
+Try GET/ greet/{name} with a name 'loopback' and language 'en'
+You will see a result object with a timestamp field like
+
+```s
+{
+  "timestamp": "2019-10-09T20:24:00.294Z",
+  "language": "en",
+  "greeting": "Hello, loopback!"
+}
+```
+
+Try the same endpoint and same input immediately (before the cached result expires), you will see the same timestamp printed in the result, which infers that the result is NOT calculated again but from cache.
+Wait for 10s and try same endpoint again, you will see the timestamp changes, like (maybe use an image to point out the difference)
+
+```s
+{
+// please note the time here changes
+"timestamp": "2019-10-09T20:30:29.770Z",
+"language": "en",
+"greeting": "Hello, loopback!"
+}
+```
+
+#### Test sweep
+
+The sweep interval is set to 10s(10,000 milliseconds) too.
+Since the cache sweep runs every 10s in the background, we can observe the sweep function gets executed by turning on the debug string and restart the app.
+
+To test the cache sweeping, you can:
+
+Stop your app and run it again with debug string turning on:
+
+```s
+// exit your app
+ctrl + c
+// start it again with debug
+DEBUG=greeter-extension npm run start
+```
+
+Watch the console window to see how the sweeping function is triggered every 10s
+
+```s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+greeter-extension Sweeping cache... +10s
+```
+
```

Comment:
![image](https://user-images.githubusercontent.com/6864736/67121256-14b28b80-f1b9-11e9-90a9-81f9a558d9ed.png)

5 seconds

Reply:
Need to check why this is 5s. It should be 10s.